Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -244,14 +244,29 @@ static String originFilePath(const String& directory)
     return FileSystem::pathByAppendingComponent(directory, OriginStorageManager::originFileIdentifier());
 }
 
+void NetworkStorageManager::writeOriginToFileIf(const WebCore::ClientOrigin& origin, Function<bool()>&& condition)
+{
+    auto* manager = m_localOriginStorageManagers.get(origin);
+    if (!manager || manager->didWriteOriginToFile())
+        return;
+
+    auto originDirectory = originDirectoryPath(m_path, origin, m_salt);
+    if (originDirectory.isEmpty())
+        return;
+
+    if (!condition)
```

Comment:
This looks wrong? It checks if condition is null but doesn't actually call the function?
If I'm right, I guess it means testing is insufficient?

As a side note, since condition is run synchronously, we can probably pass it as a `const Function<bool>()>&` instead of a rvalue reference.

Reply:
oooops. I should add a test case to verify there is origin file if the condition is true. Will change to const Function<bool>()>& too.